volatility/framework/symbols/windows/extensions/network.py

```diff
@@ -0,0 +1,208 @@
+from volatility.framework.objects import Array
+from volatility.framework.renderers import conversion
+from volatility.framework.symbols.wrappers import Flags
+from volatility.framework import renderers
+from typing import Dict, Tuple
+
+vollog = logging.getLogger(__name__)
+
+def inet_ntop(address_family: int, packed_ip: Array) -> str:
+
+    def inet_ntop4(packed_ip: Array) -> str:
+
+        if not (isinstance(packed_ip, list) or isinstance(packed_ip, Array)):
+            raise TypeError("must be Array, not {0}".format(type(packed_ip)))
+        if len(packed_ip) != 4:
+            raise ValueError("invalid length of packed IP address string")
+        return "{0}.{1}.{2}.{3}".format(*[x.to_bytes(1, "little")[0] for x in packed_ip])
+
+    def inet_ntop6(packed_ip) -> str:
+        if not (isinstance(packed_ip, list) or isinstance(packed_ip, Array)):
+            raise TypeError("must be Array, not {0}".format(type(packed_ip)))
+
+        if len(packed_ip) != 16:
+            raise ValueError("invalid length of packed IP address string")
+
+        words = []
+        for i in range(0, 16, 2):
+            words.append((packed_ip[i] << 8) | packed_ip[i + 1])
+
+        # Replace a run of 0x00s with None
+        numlen = [(k, len(list(g))) for k, g in itertools.groupby(words)]
+        max_zero_run = sorted(sorted(numlen, key = lambda x: x[1], reverse = True), key = lambda x: x[0])[0]
+        words = []
+        for k, l in numlen:
+            if (k == 0) and (l == max_zero_run[1]) and not (None in words):
+                words.append(None)
+            else:
+                for i in range(l):
+                    words.append(k)
+
+        # Handle encapsulated IPv4 addresses
+        encapsulated = ""
+        if (words[0] is None) and (len(words) == 3 or (len(words) == 4 and words[1] == 0xffff)):
+            words = words[:-2]
+            encapsulated = inet_ntop4(packed_ip[-4:])
+        # If we start or end with None, then add an additional :
+        if words[0] is None:
+            words = [None] + words
+        if words[-1] is None:
+            words += [None]
+        # Join up everything we've got using :s
+        return ":".join(["{0:x}".format(w) if w is not None else "" for w in words]) + encapsulated
+
+    if address_family == socket.AF_INET:
+        return inet_ntop4(packed_ip)
+    elif address_family == socket.AF_INET6:
+        return inet_ntop6(packed_ip)
+    raise socket.error("[Errno 97] Address family not supported by protocol")
+
+# Python's socket.AF_INET6 is 0x1e but Microsoft defines it
+# as a constant value of 0x17 in their source code. Thus we
+# need Microsoft's since that's what is found in memory.
+AF_INET = 2
+AF_INET6 = 0x17
+
+# String representations of INADDR_ANY and INADDR6_ANY
+inaddr_any = inet_ntop(socket.AF_INET, [0] * 4)
+inaddr6_any = inet_ntop(socket.AF_INET6, [0] * 16)
+
+# copied to here as symbol space does not appear to support custom Enums
+TCP_STATE_ENUM = {
```

Comment:
Yep, I'd still like it these could be in the JSON files, rather than as code.  There's a whole section for Enums in there, and then they can be accessed based on name, etc.  Send me a message if you need more help details around this (and I might write some documentation about it at the same time, if we're lucky).  5:)

Reply:
fixed